keel-core/src/main/kotlin/com/netflix/spinnaker/keel/lifecycle/LifecycleEvent.kt

```diff
@@ -0,0 +1,49 @@
+package com.netflix.spinnaker.keel.lifecycle
+
+import com.netflix.spinnaker.keel.api.artifacts.DeliveryArtifact
+import com.netflix.spinnaker.keel.lifecycle.LifecycleEventScope.PRE_DEPLOYMENT
+import com.netflix.spinnaker.keel.lifecycle.LifecycleEventType.BAKE
+import java.time.Instant
+
+/**
+ * A lifecycle event is emitted for an artifact version.
+ *
+ * Events are identified by the artifact, version, scope, type, and id.
+ * We create an id for the artifact by calling [artifact.toLifecycleEventId()].
+ *
+ * [artifactVersion] is the version of the artifact that this event pertains to.
+ *
+ * [scope] and [type] control where the event is shown, right now there is only one of each.
+ * [id] is a caller-provided id: if the caller wants to be able to update the event,
+ *  it should not be random. If the event is fire and forget, the id can be null.
+ * [link] stores a link that will be surfaced in a clickable button.
+ * [text] is the text for the even that is shown to the user.
+ */
+interface LifecycleEvent {
```

Comment:
should we add here `createdBy` or `author` or something that will tell us who initiated this event?

Reply:
These will all be generated in code, and I think the "type" should tell us who initiated them (like bake events will be initiated by the image handler)

Reply:
for the bakes it makes sense, but for builds or other stuff? we might need this information? (I'm not sure, just trying to think if something is missing).

Reply:
Resolving for now - we may need to expand this event in the future, and that's okay.